www/build.sh

```diff
@@ -58,46 +58,43 @@ find www/build/builtins -type f -name 'index.html' -exec sed -i 's!</nav>!<div c
 # cleanup files that could have stayed behind if the script failed
 rm -rf roc_nightly roc_releases.json
 
-echo 'Fetching latest roc nightly...'
-
 # to prevent GitHub from rate limiting netlify servers
 if ! [ -v GITHUB_TOKEN_READ_ONLY ]; then
-  curl https://api.github.com/repos/roc-lang/roc/releases > roc_releases.json
+  echo 'Building tutorial.html from tutorial.md...'
+  mkdir www/build/tutorial
+
+  cargo run --release --bin roc run www/generate_tutorial/src/tutorial.roc -- www/generate_tutorial/src/input/ www/build/tutorial/
 else
+  echo 'Fetching latest roc nightly...'
+  
+  # we assume that we're on a netlify server if GITHUB_TOKEN_READ_ONLY is set
```

Comment:
I've made a bunch of changes in this file, this is just to make this script work easily on NixOS.